src/Command/CleanupStringsCommand.php

```diff
@@ -325,4 +342,79 @@ protected function purifySessionDescription(OutputInterface $output)
         $output->writeln('');
         $output->writeln("<info>{$cleaned} Session Descriptions updated.</info>");
     }
+
+    /**
+     * @param OutputInterface $output
+     */
+    public function correctLearningMaterialLinks(OutputInterface $output): void
+    {
+        $cleaned = 0;
+        $offset = 0;
+        $failures = [];
+        $limit = self::QUERY_LIMIT;
+        $total = $this->learningMaterialManager->getTotalLearningMaterialCount();
+        $progress = new ProgressBar($output, $total);
+        $progress->setRedrawFrequency(208);
+        $output->writeln("<info>Starting cleanup of learning material links...</info>");
+        $progress->start();
+        do {
+            $materials = $this->learningMaterialManager->findBy([], ['id' => 'ASC'], $limit, $offset);
+            /* @var LearningMaterialInterface $material */
+            foreach ($materials as $material) {
+                $original = $material->getLink();
+                if (null === $original || '' === trim($original)) {
+                    continue;
+                }
+                try {
+                    $fixed = $this->fixLink($original);
+                    if ($original !== $fixed) {
+                        $cleaned++;
+                        $material->setLink($fixed);
+                        $this->learningMaterialManager->update($material, false);
+                    }
+                } catch (Exception $e) {
+                    $failures[] = [$material->getId(), $original, $e->getMessage()];
+                }
+                $progress->advance();
+            }
+
+            $offset += $limit;
+            $this->em->flush();
+            $this->em->clear();
+        } while (count($materials) === $limit);
+        $progress->finish();
+        $output->writeln('');
+        $output->writeln("<info>{$cleaned} learning material links updated, " . count($failures) . " failures.</info>");
+        $output->writeln('');
+        if (! empty($failures) && $output->isVerbose()) {
+            $table = new Table($output);
+            $table->setHeaders(['Learning Material ID', 'Link', 'Error Message']);
+            $table->setRows($failures);
+            $table->render();
+        }
+    }
+
+    /**
+     * @param string $link
+     * @return string|null
+     * @throws TransportExceptionInterface
+     */
+    protected function fixLink(string $link): ?string
+    {
+        $fixed = trim($link);
+        if (preg_match(';^(http|ftp)s?://;i', $fixed)) {
```

Comment:
There are a few links currently with `http://https://` (double protocol). I think this should clean those up as well if possible.

Reply:
edge case handling added.